utils/test_matrix/test_matrix.go

```diff
@@ -0,0 +1,130 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"golang.org/x/tools/go/packages"
+)
+
+const flowPackagePrefix = "github.com/onflow/flow-go/"
+const ciMatrixName = "dynamicMatrix"
+
+// testMatrix represents a single GitHub Actions test matrix combination that consists of a name and a list of flow-go packages associated with that name.
+type testMatrix struct {
+	Name     string `json:"name"`
+	Packages string `json:"packages"`
+}
+
+// Generates a list of packages to test that will be passed to GitHub Actions
+func main() {
+	if len(os.Args) == 1 {
+		fmt.Fprintln(os.Stderr, "must have at least 1 package listed")
+		return
+	}
+
+	allFlowPackages := listAllFlowPackages()
+
+	targetPackages, seenPackages := listTargetPackages(os.Args[1:], allFlowPackages)
+
+	restPackages := listRestPackages(allFlowPackages, seenPackages)
```

Comment:
nit: Can we remove the usage of **_rest_** so that it's not confused with HTTP REST, at a quick glance it would look like these are some type of REST API tests.

Reply:
Updated in https://github.com/onflow/flow-go/pull/3087/commits/5f661dbc0a3fc54a3b5fb67b799a70a3f9312cf8